### ISSUE TITLE
window.resolveLocalFileSystemURL not firing callback in non-root views

### ISSUE THREAD

**Issue by mmelkko** [AppGyver/steroids#783]
Related forum post: https://forums.appgyver.com/#!/steroids-bugs:windowresolvelocalfilesyst

No success or failure callback fired when trying to resolve the file system URL from views other than the root view of the app.

**Comment by doomboy00** [comment]
If you set up a setInterval around trying to resolve this it will often 'fix' itself after a couple of loops.. not sure why, but seems to work, at least in the scenario i'm using it.

**Comment by ssandison** [comment]
I'm using jquery.ajax() to check for file existence on localhost. It works but is dirtier than your hack which I will give a go.

I can't get my android scanner to work with any sandbox apps but here is a link to a reproduction:

http://sandbox.appgyver.com/sandboxes/1223

It works fine in iOS (doesn't have this issue) but like I said I can't test it in android scanner for some reason.

**Comment by ssandison** [comment]
The setInterval() fix doesn't work for me on Google Play builds (crosswalk ARM). I can use FileTransfer.download() to create the file but I can't use window.resolveLocalFilesystemURL to check for its existence.

**Comment by pre** [comment]
Another user was able to solve this problem [on the forums](https://forums.appgyver.com/#!/steroids:check-if-files-exist).

The key in his solution was `navigator.webkitPersistentStorage.requestQuota` before requesting file system access.

~~~
function requestQuota(){
  //browser-specific prefix
  window.requestFileSystem  = window.requestFileSystem || window.webkitRequestFileSystem;

  navigator.webkitPersistentStorage.requestQuota (150*1024*1024 /* 150mb */, function(grantedBytes) {
    //console.log ('Granted bytes: ', grantedBytes);
    checkfile(grantedBytes);
  }, fail);
}

function checkfile(grantedBytes) {
  window.webkitRequestFileSystem(window.PERSISTENT, grantedBytes, function(fs) {
    var checkFile = "file://" + steroids.app.absoluteUserFilesPath + "/teams/brazil/gremio.svg" ;
    console.log( "checking file: " + checkFile );
    fs.root.getFile( checkFile, {create: false},
    function gotFileEntry() {
      console.log( "file: " + checkFile + " exists" );
    }, downloadFiles );
  }, fail);
}
~~~

**Comment by ssandison** [comment]
This looks promising I shall give it a go.

**Comment by gringom** [comment]
Only had a problem with this, yes it worked and downloaded the files. But running a second time I didn't get the file exists. I continued getting "NotFoundError" as if the file didn't exist. Using a file manager I was able to see that the files are there:
/storage/emulated/0/android/data/com.appgyver.freshandroid/files/user/teams/brazil/flamengo.svg

I tested the directory with file:// , file:/ (since there already is a / before storage) before ```steroids.app.absoluteUserFilesPath``` and with nothing before it. No luck.

Any suggestions?

**Comment by gringom** [comment]
Just read the whole thread, from the beginning (forum) and saw that i'm having the same problem everyone's having. Just arrived from a different route.

**Comment by gringom** [comment]
The following code worked in both cases (file existing and not existing). From here on I was able to download the files when necessary and "do nothing" if they have already been downloaded before.

```document.addEventListener("deviceready", onDeviceReady, false);
function onDeviceReady() {
  window.requestFileSystem(LocalFileSystem.PERSISTENT, 0, null, fail);
  window.resolveLocalFileSystemURL("file://" + steroids.app.absoluteUserFilesPath + "/teams/brazil/flamengo.svg", onResolveSuccess, noFile);
}

function onResolveSuccess(fileEntry) {
  window.myFileExists = true ;
  requestQuota();
}

function noFile(e){
  window.myFileExists = false ;
  requestQuota();
}```

**Comment by gringom** [comment]
Just to add on this...
The workaround has failed after further testing.
It's like, sometimes (can't pin point when or why) the callback function isn't fired.

**Comment by gringom** [comment]
Debugging a bit more I realised that the few times the callback function isn't called in the console log I get
View http://localhost/app/argentina/gallery.html is not started, wont register channel 
supersonic.js:30294

**Comment by PeanutButterCo** [comment]
+1 waiting for this to be solved to release app

**Comment by harsaharsa** [comment]
The latest Android Edge build (v4.0.4-edge9) should help significantly with Cordova callbacks not firing. Can anyone verify if the fix solves this issue also?

**Comment by PeanutButterCo** [comment]
very good news, we actually have a interval set to retry window.resolveLocalFileSystemURL if it misses. Normally it takes between 1 and 2 tries, maximum 3, but it will be nice to remove this workaround. 

I imagine @harsaharsa that you mean v4.0.5-edge9 and not v4.0.4-edge9

**Comment by harsaharsa** [comment]
Yup, sorry! v4.0.5-edge9.

**Comment by JugnuPathak** [comment]
sitll not wolrking

**Comment by JugnuPathak** [comment]
Im not able to use window.resolveLocalFileSystemURL with camera returned image

**Comment by harsaharsa** [comment]
Should be fixed for good in 4.0.6-edge2, please give it a try and tell us if the issue is still there!
